25 add posts count to category list

gilly7 opened the pull request:
## Add Post Count to Category List and Add Favicon

## Changes Included:

### 1. Add Post Count to Category List
- **Backend**: Modified `category_routes.py` to calculate and include the post count for each category when listing categories.
- **Frontend**: Updated `/app/page.tsx` and `/app/categories/page.tsx` to fetch and display the post count alongside each category in the list view.
- **Models**: Extended the `CategoryResponse` model in `models.py` to include `post_count`.
Below is the updated UI with post count on category list

![image](https://github.com/user-attachments/assets/0a793d85-5d42-42c9-a203-83350473bdab)
![image](https://github.com/user-attachments/assets/faa58080-68f3-4d17-9b23-ad5e378bb217)


### 2. Add Favicon to the Project
-  This PR addresses the issue to add a favicon, closes #19.

gilly7 committed:
fix(utils): format timestamps to user-friendly date strings

gilly7 committed:
fix(utils): format timestamps to user-friendly date strings

gilly7 committed:
Add posts count to category list

etowett reviewed (changes requested):
@gilly7 looks like you made these changes without considering the latest changes in main, can you rebase your changes?

Take a look at the screenshot, you need to fix that first

![Screenshot 2025-01-06 at 09 12 53](https://github.com/user-attachments/assets/10fbb13a-accf-492f-818c-42ea87cf780e)

gilly7 committed:
Merge branch 'main' of https://github.com/Korirhillary/inkhive into 25-add-posts-count-to-category-list

gilly7 committed:
add favicon

etowett commented on frontend/app/layout.tsx:
```diff
@@ -16,6 +17,9 @@ export default function RootLayout({
   return (
     <html lang="en">
+       <Head>
+       <link rel="icon" href="/favicon.jpg" type="image/jpeg"/>
```
We do not need to do this, adding favicon file to the public directory should take care of this.

etowett commented on frontend/app/layout.tsx:
```diff
@@ -16,6 +17,9 @@ export default function RootLayout({
     <html lang="en">
+       <Head>
+       <link rel="icon" href="/favicon.jpg" type="image/jpeg"/>
+      </Head>
```
Please remove this head section.

gilly7 replied:
removed the head section on this commit.  https://github.com/Korirhillary/inkhive/pull/30/commits/747de93ffc25032bc768c746f577be3c1834b1cc

etowett reviewed (commented):
Good stuff.

One observation: Looks like the backend doesn't currently return post count, but we are using it.

Can you update that? Or please add a screenshot of how this looks like in the UI?

gilly7 committed:
feat: Add post count to category list in backend and frontend

- Updated CategoryResponse model to include post_count
- Modified list_categories endpoint to calculate and include post count
- Updated frontend to display post count in CategoryPage component

gilly7 commented:
Below is the updated UI with post count on category list
![image](https://github.com/user-attachments/assets/9b144eec-a6b0-45ad-956c-d18b7055afe2)
![image](https://github.com/user-attachments/assets/5faf3c50-e386-40e8-bbd1-97f41530b352)

gilly7 committed:
remove head section for favicon

gilly7 committed:
remove unnecessary comments

etowett commented on backend/category_routes.py:
```diff
@@ -48,6 +46,12 @@ def list_categories(
         .all()
     )
 
+    # Adding post count for each category
```
These comments are not necessary

etowett commented on backend/category_routes.py:
```diff
@@ -48,6 +46,12 @@ def list_categories(
+    # Adding post count for each category
+    for category in categories:
+        post_count = db.query(Post).filter(Post.category_id == category.id).count()
+        # Here you're adding post_count dynamically
```
Same

etowett commented on frontend/app/layout.tsx:
```diff
@@ -8,6 +8,7 @@ import { SessionProvider } from "next-auth/react";
 import React from "react";
 import AppHeader from "./AppHeader";
 import theme from "./theme";
+import Head from "next/head";
```
This is not necessary, please remove

etowett commented on frontend/app/lib/api.ts:
```diff
@@ -54,4 +54,4 @@ export const deleteCategory = async (id: number) => {
 export const getPosts = async () => {
   return fetchWithAuth("/posts");
-};
+};
```
This change is not necessary, do not remove line break here

etowett commented on frontend/app/page.tsx:
```diff
@@ -138,4 +138,4 @@ export default function Home() {
     </Grid>
   );
-}
+}
```
Line break please

gilly7 replied:
I've made the following updates based on the feedback:

- Removed the unnecessary import of `Head` from `next/head` in `layout.tsx`.
- Kept the existing line break structure in `api.ts` for the `getPosts` function. 
- Added a line break at the end of the `Home` function for consistency.
commit : https://github.com/Korirhillary/inkhive/pull/30/commits/44943bbdace007fdfa1da8569271e94fa760f6c6

etowett reviewed (approved):
Good stuff, check the comments then you can merge

gilly7 committed:
formatting adjustments
- Preserved line break in api.ts for getPosts function
- Added line break at the end of Home function for consistency
-removed unnecessary comments
